plugins/processors/t128_rate/t128_rate.go

```diff
@@ -0,0 +1,164 @@
+package rename
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+const sampleConfig = `
+	## If more than this amount of time passes between data points, the
+	## previous value will be considered old and the rate will be recalculated
+	## as if it hadn't been seen before. A zero expiration means never expire.
+	# expiration = "0s"
+
+	## For the fields who's key/value pairs don't match, should the original
+	## field be removed?
+	# remove-original = true
+
+[processors.t128_rate.fields]
+	## Replace fields with their rates, renaming them if indicated
+	# "/rate/metric" = "/total/metric"
+	# "/inline/replace" = "/inline/replace"
+`
+
+type T128Rate struct {
+	Fields         map[string]string `toml:"fields"`
+	Expiration     internal.Duration `toml:"expiration"`
+	RemoveOriginal bool              `toml:"remove-original"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	targetFields map[string]target
+	cache        map[uint64]map[string]observedValue
+}
+
+type target struct {
+	key           string
+	matchesSource bool
+}
+
+type observedValue struct {
+	value   float64
```

Comment:
So if the metric-id's value is not `float64` I'm assuming the implementation will just convert it for caching purposes? Is this the most common type or I guess can/should this be `uint64` given that this is for 128T metrics. Not sure what the additional overhead of conversion is (hopefully minimal?)

Reply:
I was using the conversion based on what `basicstats` does. I believe you're correct that we could use a `uint64` for this instead. You're right that it shouldn't matter much. I think the flexibility of being able to accommodate floats in the future is worth leaving it this way.